getting_started/step_by_step/resources.rst

```diff
@@ -158,12 +158,201 @@ Freeing resources
+
+.. tab::
+  ..code-tab:: gdscript GDScript
+    # bot_stats.gd
+    extends Resource
+    export(int) var health
+    export(Resource) var sub_resource
+    export(Array, String) var str_array
+
+    func _init(p_health = 0, p_sub_resource = null, p_str_array = []):
+        health = p_health
+        sub_resource = p_sub_resource
+        str_array = p_str_array
+
+    # bot.gd
+    extends KinematicBody
+    export(Resource) var stats
+
+    const BotStats = preload("res://bot_stats.gd")
+
+    func _init():
+        if not stats:
+            stats = BotStats.new(10)
+    
+    func _ready():
+        print(stats.health) # prints '10'
+  ..code-tab:: csharp
+        // BotStats.cs
+        using System;
+        using Godot;
+
+        public class BotStats : Resource
+        {
+            [Export]
+            public int health;
+
+            [Export]
+            public Resource subResource;
+
+            [Export]
+            public String[] strArray;
+
+            public override void _Init(int pHealth = 0, Resource pSubResource = null, String[] pStrArray = [])
+            {
+                health = pHealth;
+                subResource = pSubResource;
+                strArray = pStrArray;
+            }
+        }
+
+        // Bot.cs
+        using System;
+        using Godot;
+
+        public class Bot : KinematicBody
+        {
+            [Export]
+            public BotStats stats;
+
+            public override void _Init()
+            {
+                if (!stats)
+                {
+                    stats = (ResourceLoader::load("BotStats.cs") as BotStats).new(10);
+                }
+            }
+
+            public override void _Ready()
+            {
+                Godot.print(stats.health);
+            }
+        }
+
+.. note::
+
+   Resource scripts are similar to Unity's ScriptableObjects. The Inspector 
+   provides built-in support for custom resources. If desired, users can even 
+   design their own Control-based tool scripts and combine them with an
+   :ref:`EditorPlugin <class_EditorPlugin>` to create custom visualizations
+   and editors for their data.
+
+   Unreal Engine 4's DataTables and CurveTables are also easy to recreate with
+   Resource scripts. DataTables are a String mapped to a custom struct, similar
+   to a Dictionary mapping a String to a secondary custom Resource script.
+
+    .. tabs::
+      .. code-tab:: gdscript GDScript
+
+        extends Resource
+
+        const BotStats = preload("bot_stats.gd") 
+
+        func _init():
+            var dict = {
+                "GodotBot": BotStats.new(10), # creates instance with 10 health
+                "DifferentBot": BotStats.new(20) # a different one with 20 health
+            }
+            print(dict)
+      .. code-tab:: csharp
+        using Godot;
+
+        public class BotStatsTable : Resource
+        {
+            private Godot.Dictionary<String, BotStats> stats = new Godot.Dictionary<String, BotStats>();
```

Comment:
Use underscore prefix for private members http://docs.godotengine.org/en/latest/getting_started/scripting/c_sharp/c_sharp_style_guide.html#naming-conventions